### PR TITLE
Use characters instead of `u32` in confusable map

### PR DESCRIPTION
**Pull request by charliermarsh**

### CODE REVIEW

**Comment by charliermarsh** [comment]
I don't feel strongly about this but it came up once before. Sadly it really buffs my commit stats.

File: crates/ruff_linter/src/rules/ruff/rules/ambiguous_unicode_character.rs [hunk 1]
```diff
@@ -163,7 +163,7 @@ pub(crate) fn ambiguous_unicode_character(
                     let candidate = Candidate::new(
                         TextSize::try_from(relative_offset).unwrap() + range.start(),
                         current_char,
-                        char::from_u32(representant).unwrap(),
+                        representant,
```
Comment (charliermarsh) [inline crates/ruff_linter/src/rules/ruff/rules/ambiguous_unicode_character.rs:166]: The motivation for doing this _now_ is that we can avoid this unwrap.

**Review by MichaReiser — APPROVED** [review]
I like being explicit about type.

**Comment by zanieb** [comment]
@charliermarsh why did this cause ecosystem changes?

**Comment by charliermarsh** [comment]
@zanieb - I think those are incorrect -- those are from https://github.com/astral-sh/ruff/pull/4430 which I tagged you on.

**Pull request merged**